tests/unit/test_sql_sensor.py

```diff
@@ -106,5 +108,45 @@ def tearDownClass(cls) -> None:
         cls._get_records_patch.__exit__(None, None, None)
 
 
+class ClickHouseLegacySqlSensorTestCase(unittest.TestCase):
+    @mock.patch(
+        'airflow_clickhouse_plugin.ClickHouseSqlSensor'
+            '._ClickHouseSqlSensor__poke_clickhouse',
+    )
+    @mock.patch('airflow.sensors.sql_sensor.SqlSensor.poke')
+    @mock.patch('airflow.sensors.sql_sensor.SqlSensor._get_hook', create=True)
+    def test_get_hook_defined(
+            self,
+            _: mock.MagicMock,  # force creation of _get_hook
+            sql_sensor_poke_mock: mock.MagicMock,
+            clickhouse_sensor_poke_mock: mock.MagicMock,
+    ):
+        """ Test for case when ``SqlSensor._get_hook`` method is present. """
+        op = ClickHouseSqlSensor(task_id='_', sql='')
+        context = dict(some_test_context=True)
+        op.poke(context)
+        sql_sensor_poke_mock.assert_called_once_with(context)
+        clickhouse_sensor_poke_mock.assert_not_called()
+
+    @mock.patch(
+        'airflow_clickhouse_plugin.ClickHouseSqlSensor'
+            '._ClickHouseSqlSensor__poke_clickhouse',
+    )
+    @mock.patch(
+        'airflow_clickhouse_plugin.sensors.clickhouse_sql_sensor.SqlSensor',
+    )
+    def test_get_hook_missing(
+            self,
+            sql_sensor_mock: mock.MagicMock,
+            clickhouse_sensor_poke_mock: mock.MagicMock,
+    ):
+        """ Test for case when ``SqlSensor._get_hook`` method is present. """
+        op = ClickHouseSqlSensor(task_id='_', sql='')
+        delattr(sql_sensor_mock, '_get_hook')
+        op.poke(context=None)
+        sql_sensor_mock.poke.assert_not_called()
+        clickhouse_sensor_poke_mock.assert_called_once_with()
+
```

Comment:
Checked on 1.10.10 and it seems that `delattr` actually doesn't delete the `_get_hook` method.
Got this error:
`AssertionError: Expected '_ClickHouseSqlSensor__poke_clickhouse' to be called once. Called 0 times.`

Reply:
Actually I don't sure does this test should work on 1.10.10 😳

Reply:
Yeah, thanks, I did not test it myself but after you comment I have updated the test (please see the new commit) and checked it over SqlSensor implementation with `_get_hook` method present (like for versions ≥1.10.7) and it works.

So I have tested for Airflow 1.10.6 and «patched» 1.10.6 (with `_get_hook` added). Please check it yourself against 1.10.10 too.